generators/tracegen/src/main/scala/Tile.scala

```diff
@@ -29,6 +34,180 @@ class TraceGenTile(val id: Int, val params: TraceGenParams, q: Parameters)
   override lazy val module = new TraceGenTileModuleImp(this)
 }
 
+class BoomTraceGenTile(val id: Int, val params: TraceGenParams, q: Parameters)
+  extends BaseTile(params, SynchronousCrossing(), HartsWontDeduplicate(params), q) {
+  val boom_params = p.alterMap(Map(TileKey -> BoomTileParams(
+    dcache=params.dcache,
+    core=BoomCoreParams(nPMPs=0, numLdqEntries=32, numStqEntries=32, useVM=false))))
+  val dcache = params.dcache.map {
```

Comment:
Why use map here if you're just going to call get and don't use the actual config? Just create the lazy module unconditionally.